cabal-install/Distribution/Client/Dependency/Modular/Linking.hs

```diff
@@ -401,7 +428,7 @@ verifyFlag' (FN (PI pn i) fn) lg = do
         vals  = map (`M.lookup` vsFlags vs) flags
     if allEqual (catMaybes vals) -- We ignore not-yet assigned flags
       then return ()
-      else conflict ( S.fromList (map F flags) `S.union` lgConflictSet lg
+      else conflict ( CS.fromList (map F flags) `CS.union` lgConflictSet lg
                     , "flag " ++ show fn ++ " incompatible"
                     )
```

Comment:
Because if I link Q.A to Q'.A, then it doesn't make sense to allow Q.A-flag to be true but Q'.A-flag to be false.